simulator/resourcewatcher/resourcewatcher_test.go

```diff
@@ -0,0 +1,484 @@
+package resourcewatcher_test
```

Comment:
Hm, it doesn't make sense to change test to be in separate package only for solving import cycles.

I'm not sure but seems like `StreamWriter` is one cause...? (Because it uses `WatchEvent` in its definition.)
If so, one idea to solve this is just to create another package for `StreamWriter` and move the related implementation into that new package.